zbuf/scanner.go

```diff
@@ -163,3 +187,42 @@ func (n *namedScanner) Pull(done bool) (Batch, error) {
 	}
 	return b, err
 }
+
+type MultiScanner struct {
+	scanners []Scanner
+	progress Progress
+	current  Scanner
+}
+
+var _ Scanner = (*MultiScanner)(nil)
+
+func NewMultiScanner(scanners ...Scanner) *MultiScanner {
+	return &MultiScanner{
+		scanners: scanners,
+	}
+}
+
+func (m *MultiScanner) Pull(done bool) (Batch, error) {
```

Comment:
Nits: No point exporting this type. And call the function MultiScanner for symmetry with zio.ConcatReader and zio.MultiWriter.
```suggestion
func MultiScanner(scanners ...Scanner) Scanner {
	return &multiScanner{
		scanners: slices.Clone(scanners),
	}
}

type multiScanner struct {
	scanners []Scanner
	progress Progress
	current  Scanner
}

func (m *multiScanner) Pull(done bool) (Batch, error) {
```